### DISCUSSION TITLE
Adopting a whitelist-based directory naming convention (maybe similar to lutris's)

### DISCUSSION THREAD

**Discussion by CartoonFan** [tkashkin/GameHub#636] [Implemetation details]
Hello, there! @tkashkin,  now that we can finally use game titles for folders outside of the main game directory, I was wondering if it would be possible to have directory names that are bit more POSIX-friendly. Although their directory structure is similar (currently install_dir/game_name/provider/files, from what I can tell), Lutris seems to be using lowercase characters, replacing spaces with dashes, and stripping directory names of nonstandard characters. I think it'd be helpful to have a similar naming convention (possibly whitelist-based?) as well. An added bonus of using Lutris's naming scheme is increased compatibility (not having to re-download files and such), but I don't think that's a strict requirement (for example, I think underscores are better than dashes, but that's just me).

Anyway, I'd love to hear people's thoughts about this! It's not necessarily a **huge** thing, but I think it's a nice thing to have, for the reasons listed above. Thanks!

**Comment by tkashkin** [comment]
GameHub already strips the directory names of installed games to improve compatibility as some games don't run with spaces or other weird characters in paths.

Collection paths are currently not stripped (other than replacing `:`) because the original idea of Collections is to store the downloaded DRM-free installers in a user-friendly way while keeping the original human-readable game names. Stripping is not required here anyway because installers usually don't care about the path they're executed from.

I'm not against changing this and possibly making it more interoperable with other software. Other opinions and more details about what Lutris, Heroic and other related software does would be helpful.

I guess the easiest option right now is to keep the defaults as they are and add a separate variable like `shortname`, `slug` or whatever else it should be called.

**Reply by CartoonFan** [reply]
That's fine, I guess. As long as the directories can be accessed by the tools, I'm OK with that. I don't have an Epic account, and I can't get Heroic to work with GOG either, so I'll just show some  GameHub/Lutris title comparisons.

```
GameHub: Far Cry® 2 - Fortune's Edition
Lutris: far-cry-2
```
```
GameHub: Beyond Good & Evil™
Lutris: beyond-good-evil
```
```
GameHub: Outcast 1.1
Lutris: outcast-11
```
```
GameHub: Baldur's Gate - The Original Saga
Lutris: baldurs-gate-the-original-saga
```
I think I'll try going with GameHub's names at the moment, and see if I can get used to them. Thanks again for responding so quickly!
